src/Microsoft.AspNetCore.Build/targets/Compile.targets

```diff
@@ -0,0 +1,21 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+    <!-- Put targets attached to the 'Compile' phase of the Standard Lifecycle here. -->
+
+    <Target Name="DotNetBuild" BeforeTargets="Compile">
+        <ItemGroup>
+            <_ToBuild Include="@(Projects)" Condition="'$(BuildSourceOnly)' != 'true' Or '%(Projects.ProjectGroup)' == 'src'" />
+        </ItemGroup>
+
+        <Delete Files="%(_ToBuild.GeneratedBuildInfoFile)" Condition="Exists(%(_ToBuild.GeneratedBuildInfoFile))" />
+        <WriteLinesToFile
```

Comment:
Does it always generate this? We do this conditionally in old-build to avoid depending on git being in the path (plus the cost on dev boxes where it's pretty unnecessary).